pyro/optim/optim.py

```diff
@@ -51,12 +51,12 @@ def __call__(self, params,  *args, **kwargs):
                     state = self._state_waiting_to_be_consumed.pop(param_name)
                     self.optim_objs[p].load_state_dict(state)
 
-            # actually perform the step for the optim object
-            self.optim_objs[p].step(*args, **kwargs)
-
-            # if optim object was a scheduler, perform an actual optim step
-            if isinstance(self.optim_objs[p], torch.optim.lr_scheduler._LRScheduler):
+            if isinstance(self.optim_objs[p], torch.optim.lr_scheduler._LRScheduler) or \
```

Comment:
why is this branch necessary

Reply:
we only want to step the optimizer, not the scheduler here. this branch is if `self.optim_objs[p]` is a scheduler.